proposals/testing/NNNN-polling-confirmations.md

```diff
@@ -0,0 +1,512 @@
+  /// If the expression does not finish evaluating before the timeout is
+  /// reached, then a failure will be reported.
+  case stopsPassing
+}
+```
+
+### New Error Type
+
+A new error type, `PollingFailedError` to be thrown when the polling
+confirmation doesn't pass:
+
+```swift
+/// A type describing an error thrown when polling fails.
+public struct PollingFailedError: Error, Sendable, CustomIssueRepresentable {}
+```
+
+### New `Issue.Kind` case
+
+A new issue kind will be added to report specifically when a test fails due to
+a failed polling confirmation.
+
+```swift
+public struct Issue {
+  public enum Kind {
+    /// An issue due to a polling confirmation having failed.
+    ///
+    /// This issue can occur when calling ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-455gr``
+    /// or
+    /// ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-5tnlk``
+    /// whenever the polling fails, as described in ``PollingStopCondition``.
+    case pollingConfirmationFailed
+  }
+}
+```
+
+### New Traits
+
+Two new traits will be added to change the default values for the
+`duration` and `interval` arguments. Test authors will often want to poll for
+the `firstPass` stop condition for longer than they poll for the
+`stopsPassing` stop condition, which is why there are separate traits for
+configuring defaults for these functions.
+
+```swift
+/// A trait to provide a default polling configuration to all usages of
+/// ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-455gr``
+/// and
+/// ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-5tnlk``
+/// within a test or suite for the ``PollingStopCondition.firstPass``
+/// stop condition.
+///
+/// To add this trait to a test, use the
+/// ``Trait/pollingUntilFirstPassDefaults`` function.
+@available(_clockAPI, *)
+public struct PollingUntilFirstPassConfigurationTrait: TestTrait, SuiteTrait {
+  /// How long to continue polling for
+  public var duration: Duration?
+  /// The minimum amount of time to wait between polling attempts
+  public var interval: Duration?
+
+  public var isRecursive: Bool { true }
+}
+
+/// A trait to provide a default polling configuration to all usages of
+/// ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-455gr``
+/// and
+/// ``confirmation(_:until:within:pollingEvery:isolation:sourceLocation:_:)-5tnlk``
+/// within a test or suite for the ``PollingStopCondition.stopsPassing``
+/// stop condition.
+///
+/// To add this trait to a test, use the ``Trait/pollingUntilStopsPassingDefaults``
+/// function.
+@available(_clockAPI, *)
+public struct PollingUntilStopsPassingConfigurationTrait: TestTrait, SuiteTrait {
+  /// How long to continue polling for
+  public var duration: Duration?
+  /// The minimum amount of time to wait between polling attempts
+  public var interval: Duration?
+
+  public var isRecursive: Bool { true }
```

Comment:
You can omit the `{ true }` _implementation_ and only show that it's a `get`-able property, e.g.

```suggestion
  public var isRecursive: Bool { get }
```

But I would recommend _documenting_ that this trait type is recursively applied to children. For other trait types we've done this in the DocC for the type itself.